README.md

```diff
@@ -1,5 +1,46 @@
 [![pub package](https://img.shields.io/pub/v/lints.svg)](https://pub.dev/packages/lints)
 [![Build Status](https://github.com/dart-lang/lints/workflows/validate/badge.svg)](https://github.com/dart-lang/lints/actions?query=branch%3Amain)
 
-This repo contains official Dart lint rules.
+# Official Dart lint rules.
 
+The Dart linter is a static analyzer for identifying possible problems in your
+Dart source code. More than a hundred [linter rules][rules] are available,
+checking anything from potential typing issues, coding style, and formatting.
+
+The current `package:lints` contains the official selections of lints that the
+Dart team encourages using.
+
+Two sets of lints are available:
+
+* ***Core lints***: Lints that help identify critical issues that are likely to
+lead to problems when running or consuming Dart code. All code should pass these
+lints.
+
+* ***Recommended lints***: Lints that help identify additional issues that may
+to lead to problems when running or consuming Dart code, and lints that enforce
+writing Dart using a single, ideomatic style and format. All code is encouraged
+to pass these lints. The recommended lints include all the core lints.
+
+## How these lints are used
+
+When creating new Dart project using the [`dart create`][dart create] command,
+the 'recommended' set of lints are enabled by default.
```

Comment:
Should we change this to core or update the template?

/cc @devoncarew @mit-mit

Reply:
My vote is for updating `dart create` to push people towards the recommended set. There are a lot of good lints in there. It would also align a little better with flutter since `flutter_lints` builds on top of recommended.

Reply:
Let's keep that as a separete issue; no need to block this readme update on that. I've made the text here more general.

I'll share a doc with some analysis for making the choice regarding `dart create`.